app/main.py

```diff
@@ -1,3 +1,125 @@
+import datetime
+import json
+import math
+
+from pathlib import Path
+
+BASE_DIR = Path(__file__).resolve().parent
+
+
+class Customer:
+    def __init__(self, data: dict):
+        self.name = data["name"]
+        self.products_dict_needed = data["product_cart"]
+        self.location = data["location"]
+        self.money = data["money"]
+        self.car_consumption = data["car"]["fuel_consumption"]
+
+
+class Shop:
+    def __init__(self, data: dict):
+        self.name = data["name"]
+        self.location = data["location"]
+        self.products_dict = data["products"]
+        self.trip_costs = None
+
+
+def parse_json_file():
+    with open(BASE_DIR / "config.json", "r") as f:
+        data_dict = json.load(f)
+
+    return data_dict
+
+
+def cost_of_the_trip(
+        fuel_price: int,
+        fuel_consumption: int,
+        customer_location: list,
+        shop_location: list,
+        customer_products_needed: dict,
+        shop_products_prices: dict,
+):
+    distance = math.dist(customer_location, shop_location) * 2
+    fuel_cost = (distance * fuel_consumption) / 100 * fuel_price
+
+    trip_cost_data_dict = {"products": {}}
+    total_product_cost = 0
+
+    for product, amount in customer_products_needed.items():
+        product_cost = amount * shop_products_prices[product]
+        trip_cost_data_dict["products"][product] = (amount, product_cost)
+        total_product_cost += product_cost
+
+    trip_cost_data_dict["total_product_cost"] = total_product_cost
+    trip_cost_data_dict["the_whole_cost"] = total_product_cost + fuel_cost
+
+    return trip_cost_data_dict
+
+
 def shop_trip():
```

Comment:
i think , this function is very long, may split her? for exemple, from $99-112 line, this can be in class Shop, like method "print a check", what do you thinking?

Reply:
I have decided to implement all prints in one fuction, classes used only for creating instances . If you think that is bad implementaition, i will remake it

Reply:
yes, i think, should, refactor this,

Reply:
ok

Reply:
i think, print a check its responsibility class Shop

Reply:
agree

Reply:
i will refector